src/core/lib/promise/activity.cc

```diff
@@ -124,7 +124,7 @@ void FreestandingActivity::DropHandle() {
 
 Waker FreestandingActivity::MakeNonOwningWaker() {
   mu_.AssertHeld();
```

Comment:
Unrelated to this patch: Can we replace `AssertHeld()` with `ABSL_EXCLUSIVE_LOCKS_REQUIRED` given the cost of `AssertHeld` at runtime?

Reply:
I'll give it a shot again... compiler was getting messed up last time I tried it.

That said, I think longer term this class goes away and we make everything `Party`

Reply:
(it'll be another PR though)